sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -160,6 +169,17 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
   private def tryParseDouble(field: String): DataType = {
     if ((allCatch opt field.toDouble).isDefined || isInfOrNan(field)) {
       DoubleType
+    } else {
+      tryParseTimestampNTZ(field)
```

Comment:
When I supported timestamp inference in JSON https://github.com/apache/spark/pull/23201, I had to introduce new option in https://github.com/apache/spark/pull/23455, and disable the inference by default in https://github.com/apache/spark/pull/28966 because such inference slowed down user queries. If you are sure that  is not the case, please, extend CSV benchmarks to proof the performance doesn't degrade.

Reply:
I am not working on JSON, this is a CSV data source.

Reply:
I showed JSON as an example because I guess CSV has similar issue.